app/i18n.py

```diff
@@ -0,0 +1,27 @@
+import gettext
+import os
+from .settings import I18N_DIR
+
+
+class TranslationStore:
+    def lang(lang_code=None):
+        defualt_lang = "en"
+        if lang_code is None:
+            return defualt_lang
+        else:
+            os.environ["LANGUAGE"] = lang_code
+            available_lang = gettext.find("knowledge-panel", "i18n")
+            if available_lang is None:
+                return defualt_lang
+            else:
+                return lang_code
+
+    def load():
+        lang_translations = gettext.translation(
+            "knowledge-panel",
+            localedir=str(I18N_DIR),
+            languages=["fr"],
+        )
+        lang_translations.install()
+        _ = lang_translations.gettext
+        return _
```

Comment:
First:
1. you have a typo in i18n.py: "defualt_lang" instead of "default_lang"
2. your methods needs to get self as first parameter !

Now I would better do this:
- have this class be instanciated once, at module level:
  at the end, just add `translation_store = TranslationStore()`
  from then on you never call the class but your instance (this is known as "singleton pattern")
- have a method  load_all to load languages and put every lang_translations objects in a dict (a class member) and call it just after translation_store instantiation, or better at [server startup](https://fastapi.tiangolo.com/advanced/events/). The list of language is either a configuration parameter, or determined by looking at existing folders in I18N_DIR.
- the lang method should then just return the gettext method of the corrensponding lang_translation object (which is the dict at the entry corresponding to the language)

So in you API method, you can just use:
```python
from .i18n import translation_store
...
def xxx(...):
   _ =  translation_store.lang(lang_code)
   msg = _("This will be translated")
```

Although what I find cumbersome is that you will have to carry this lang or _ variable all around.

On smarter solution could be to use a context manager with a [ContextVar](https://docs.python.org/3/library/contextvars.html) such that you could invoke it all around your code, magically.

1. in i18n.py you need a ContextVar current_lang which default value is "en"
2. write a context manager LangManager that sets this value to a value provided by the user
3. have get_translation (aka _) be a new function which read this variable, get the right translation lang (by mechanism described above) according to current_lang and use it's gettext function
4. in your api you know write:
```
from .i18n import LangManager
from .i18n import translate as _

def xxx(...):
   with LangManager("fr"):
       call_function()
       ...
       msg = _("my message")
 
```

and in inner functions, just use `from .i18n import translate as _` without carrling, the current_lang should be set.